File: app/types/storage.ts
export type Image = {
    large: string
    small: string
}

export type UploadImageResponse = Image

export type DeleteImageRequest = {
    url: string
}

export type DeleteImageResponse = {
    message: string
    large?: string | null
}

export type DeleteStorageResponse = {
    message: string
    id?: string | null
}

File: services/shared/contracts/storage.go
package contracts

import (
	"shopping-list/shared/models"
)

type UploadImageResponse models.Image

type DeleteImageRequest struct {
	URL string `json:"url" validate:"required"`
}

type DeleteImageResponse struct {
	Message string `json:"message"`
	Large   string `json:"large,omitempty"`
}

type DeleteStorageResponse struct {
	Message string `json:"message"`
	Id      string `json:"id,omitempty"`
}
